onebusaway-transit-data-federation/src/main/java/org/onebusaway/transit_data_federation/impl/realtime/gtfs_realtime/GtfsRealtimeTripLibrary.java

```diff
@@ -545,8 +545,8 @@ private int computeArrivalTime(StopTimeEntry stopTime,
       return stopTime.getArrivalTime() + arrival.getDelay();
     if (arrival.hasTime())
       return (int) (arrival.getTime() - serviceDate / 1000);
-    throw new IllegalStateException(
-        "expected arrival delay or time for stopTimeUpdate " + stopTimeUpdate);
+	
```

Comment:
I'd recommend a comment here indicating that if we've gotten this far the GTFS-rt data is bad (i.e., violates the spec), and preferably log it so data isn't just disappearing and there is some indication of the bad data externally.